File: backend/src/main/java/com/fdsa/altairis/service/BookingsService.java
package com.fdsa.altairis.service;

import com.fdsa.altairis.dto.request.BookingRequest;
import com.fdsa.altairis.dto.response.BookingNightResponse;
import com.fdsa.altairis.dto.response.BookingResponse;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.time.LocalDate;
import java.util.List;

public interface BookingsService {
    BookingResponse create(BookingRequest request, String user);
    BookingResponse cancel(Long bookingId, String user);
    BookingResponse getById(Long bookingId);
    Page<BookingResponse> list(Long hotelId, String status, LocalDate from, LocalDate to, Pageable pageable);
    List<BookingNightResponse> nights(Long bookingId);
}

File: frontend/src/app/services/services/bookings.service.ts
import { Injectable } from '@angular/core';
import { ApiClient } from '../../core/api/api.client';
import { PageResponse } from '../../core/api/api.types';
import { BookingCreateRequest } from '../../interfaces/request/booking.request';
import { BookingNightResponse } from '../../interfaces/response/booking-night.response';
import { BookingResponse } from '../../interfaces/response/booking.response';

@Injectable({ providedIn: 'root' })
export class BookingsService {
  constructor(private api: ApiClient) {}

  create(req: BookingCreateRequest) {
    return this.api.post<BookingResponse>('/api/bookings', req);
  }

  cancel(id: number) {
    return this.api.post<BookingResponse>(`/api/bookings/${id}/cancel`, {});
  }

  getById(id: number) {
    return this.api.get<BookingResponse>(`/api/bookings/${id}`);
  }

  list(hotelId: number, status: string | null, from: string, to: string, page: number, size: number) {
    return this.api.get<PageResponse<BookingResponse>>('/api/bookings', {
      hotelId,
      status: status ?? undefined,
      from,
      to,
      page,
      size,
    });
  }

  nights(id: number) {
    return this.api.get<BookingNightResponse[]>(`/api/bookings/${id}/nights`);
  }
}
